File: solutions/Python/1679.MaxNumberOfK-SumPairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        res: int = 0
        numsDict: Dict[int, int] = {}

        for num in nums:
            complement: int = k - num
            if numsDict.get(complement, 0) > 0:
                res += 1
                numsDict[complement] -= 1
            else:
                numsDict[num] = numsDict.get(num, 0) + 1

        return res

File: solutions/TypeScript/1679.MaxNumberOfK-SumPairs.ts
function maxOperations(nums: number[], k: number): number {
  let res: number = 0;
  const numsMap: Map<number, number> = new Map();

  for (const num of nums) {
    let complement: number = k - num;
    if (numsMap.get(complement) > 0) {
      res++;
      numsMap.set(complement, numsMap.get(complement) - 1);
    } else {
      numsMap.set(num, (numsMap.get(num) || 0) + 1);
    }
  }

  return res;
}
